controller/functionApi.go

```diff
@@ -188,3 +194,66 @@ func (a *API) FunctionLogsApiPost(w http.ResponseWriter, r *http.Request) {
 	}
 	proxy.ServeHTTP(w, r)
 }
+
+// FunctionPodLogs : Get logs for a function directly from pod
+func (a *API) FunctionPodLogs(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	fnName := vars["function"]
+	ns := vars["namespace"]
+
+	if len(ns) == 0 {
+		ns = "fission-function"
+	}
+
+	f, err := a.fissionClient.Functions(api.NamespaceDefault).Get(fnName)
+	if err != nil {
+		a.respondWithError(w, err)
+		return
+	}
+	envName := f.Spec.Environment.Name
+
+	_, clientset, err := tpr.GetKubernetesClient()
```

Comment:
It would be better to move this to the init of the api object -- so you don't have to login to kubernetes api on every logs request